chaoskube/chaoskube_test.go

```diff
@@ -288,10 +292,31 @@ func TestTerminateVictimOnNonExcludedWeekdays(t *testing.T) {
 	})
 }
 
+// TestTerminateVictimRespectsTimezone tests that victim is terminated when weekday filter doesn't match due to different timezone.
+func TestTerminateVictimRespectsTimezone(t *testing.T) {
+	timezone, err := time.LoadLocation("Australia/Brisbane")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	chaoskube := setup(t, labels.Everything(), labels.Everything(), labels.Everything(), []time.Weekday{time.Friday}, timezone, false, 2000)
+
+	// simulate that it's a Friday in our test (UTC). However, in Australia it's already Saturday.
+	chaoskube.Now = ThankGodItsFriday{}.Now
+
+	if err := chaoskube.TerminateVictim(); err != nil {
+		t.Fatal(err)
+	}
+
+	validateCandidates(t, chaoskube, []map[string]string{
+		{"namespace": "testing", "name": "bar"},
```

Comment:
How do you know which pod will be left if the selection is "random"? Is it because, by testing ahead of time with the same Seed, you know it always terminates the same victim? If so, I get that it works, but doesn't "smell" right. (What if logic in the `rand` package changes?) I would consider simply asserting that the number of pods available was reduced by 1 would also be valid.

Reply:
Good point. I was thinking, that by providing a seed I can control what's the "random" outcome. I used it to test that pods not matching the filters are really ignored (by testing that without the filter and the same seed it gets deleted).

However, for this test it doesn't matter which pod is killed. I'll change it as you suggested.

Reply:
@twildeboer Where applicable, I removed testing for particular pods in favour of testing just the number of surviving pods. Also removed defining a seed where it doesn't influence the outcome.